core/services/ardupilot_manager/mavlink_proxy/main.py

```diff
@@ -45,18 +46,21 @@
 
     args = parser.parse_args()
 
-    tool = AbstractRouter.get_interface(args.tool)()
-    logger.info(f"Starting {tool.name()} version {tool.version()}.")
-
-    if tool:
-        manager.use(tool)
-    else:
-        logger.warning("No tool selected. Falling back to the first one found")
-        manager.use(interfaces[0]())
-    manager.add_endpoints(args.output.split(":"))
-
-    logger.info(f"Command: {manager.command_line()}")
-    manager.start(args.master.split(":"))
-    while manager.is_running():
-        time.sleep(1)
-    logger.info("Done.")
+    async def main() -> None:
+        tool = AbstractRouter.get_interface(args.tool)()
+        logger.info(f"Starting {tool.name()} version {tool.version()}.")
+
+        if tool:
```

Comment:
Shouldn't this check be before the previous log message ?

Reply:
good catch. fixed